components/omega/test/ocn/EosTest.cpp

```diff
@@ -0,0 +1,374 @@
+   /// Get Eos instance to test
+   Eos *TestEos       = Eos::getInstance();
+   TestEos->EosChoice = EosType::LinearEos;
+
+   /// Create and fill ocean state arrays
+   Array2DReal SArray = Array2DReal("SArray", Mesh->NCellsAll, NVertLevels);
+   Array2DReal TArray = Array2DReal("TArray", Mesh->NCellsAll, NVertLevels);
+   Array2DReal PArray = Array2DReal("PArray", Mesh->NCellsAll, NVertLevels);
+   /// Use Kokkos::deep_copy to fill the entire view with the ref value
+   deepCopy(SArray, Sa);
+   deepCopy(TArray, Ct);
+   deepCopy(PArray, P);
+   deepCopy(TestEos->SpecVol, 0.0);
+
+   /// Compute specific volume
+   TestEos->computeSpecVol(TArray, SArray, PArray);
+
+   /// Check all array values against expected value
+   int numMismatches   = 0;
+   Array2DReal SpecVol = TestEos->SpecVol;
+   parallelReduce(
+       "CheckSpecVolMatrix-linear", {Mesh->NCellsAll, NVertLevels},
+       KOKKOS_LAMBDA(int i, int j, int &localCount) {
+          if (!isApprox(SpecVol(i, j), LinearExpValue, RTol)) {
+             localCount++;
+          }
+       },
+       numMismatches);
+
+   if (numMismatches != 0) {
+      Err++;
+      LOG_ERROR("EosTest: SpecVol Linear isApprox FAIL, "
+                "expected {}, got {} mismatches",
+                LinearExpValue, numMismatches);
+   }
+   if (Err == 0) {
+      LOG_INFO("EosTest SpecVolCalc Linear: PASS");
+   }
+
+   return Err;
+}
+
+/// Test Linear EOS calculation with vertical displacement
+int testEosLinearDisplaced() {
+   int Err          = 0;
+   const auto *Mesh = HorzMesh::getDefault();
+   /// Get Eos instance to test
+   Eos *TestEos       = Eos::getInstance();
+   TestEos->EosChoice = EosType::LinearEos;
+
+   /// Create and fill ocean state arrays
+   Array2DReal SArray = Array2DReal("SArray", Mesh->NCellsAll, NVertLevels);
+   Array2DReal TArray = Array2DReal("TArray", Mesh->NCellsAll, NVertLevels);
+   Array2DReal PArray = Array2DReal("PArray", Mesh->NCellsAll, NVertLevels);
+   /// Use Kokkos::deep_copy to fill the entire view with the ref value
+   deepCopy(SArray, Sa);
+   deepCopy(TArray, Ct);
+   deepCopy(PArray, P);
+   deepCopy(TestEos->SpecVolDisplaced, 0.0);
+
+   /// Compute displaced specific volume
+   TestEos->computeSpecVolDisp(TArray, SArray, PArray, KDisp);
+
+   /// Check all array values against expected value
+   int numMismatches            = 0;
+   Array2DReal SpecVolDisplaced = TestEos->SpecVolDisplaced;
+   parallelReduce(
+       "CheckSpecVolDispMatrix-linear", {Mesh->NCellsAll, NVertLevels},
+       KOKKOS_LAMBDA(int i, int j, int &localCount) {
+          if (!isApprox(SpecVolDisplaced(i, j), LinearExpValue, RTol)) {
+             localCount++;
+          }
+       },
+       numMismatches);
+
+   if (numMismatches != 0) {
+      Err++;
+      LOG_ERROR("EosTest: Linear SpecVolDisp isApprox FAIL, "
+                "expected {}, got {} with {} mismatches",
+                LinearExpValue, SpecVolDisplaced(1, 1), numMismatches);
```

Comment:
Accessing GPU array `SpecVolDisplaced` from the host.

Reply:
Certainly, `SpecVolDisplaced` usage here is an issue. But I am not sure if this is the only issue because `isApprox` is called before this logging macro and it fails with `inf` value in `SpecVolDisplaced`. So, the question here is why `SpecVolDisplaced` has an `inf` value from the beginning.